src/charm.py

```diff
@@ -5,113 +5,43 @@
 
 # Learn more at: https://juju.is/docs/sdk
 
-"""Charm the service.
-
-Refer to the following post for a quick-start guide that will help you
-develop a new k8s charm using the Operator Framework:
-
-https://discourse.charmhub.io/t/4208
-"""
+"""haproxy-operator charm file."""
 
 import logging
 import typing
 
 import ops
-from ops import pebble
 
-# Log messages can be retrieved using juju debug-log
-logger = logging.getLogger(__name__)
+from haproxy import HAProxyService
 
-VALID_LOG_LEVELS = ["info", "debug", "warning", "error", "critical"]
+logger = logging.getLogger(__name__)
 
 
-class IsCharmsTemplateCharm(ops.CharmBase):
-    """Charm the service."""
+class HAProxyCharm(ops.CharmBase):
+    """Charm haproxy."""
 
     def __init__(self, *args: typing.Any):
-        """Construct.
+        """Initialize the charm and register event handlers.
 
         Args:
-            args: Arguments passed to the CharmBase parent constructor.
+            args: Arguments to initialize the charm base.
         """
         super().__init__(*args)
-        self.framework.observe(self.on.httpbin_pebble_ready, self._on_httpbin_pebble_ready)
-        self.framework.observe(self.on.config_changed, self._on_config_changed)
+        self.haproxy_service = HAProxyService()
+        self.framework.observe(self.on.install, self._on_install)
 
-    def _on_httpbin_pebble_ready(self, event: ops.PebbleReadyEvent) -> None:
-        """Define and start a workload using the Pebble API.
+    def _on_install(self, _: typing.Any) -> None:
+        """Install the haproxy package.
 
-        Change this example to suit your needs. You'll need to specify the right entrypoint and
-        environment configuration for your specific workload.
-
-        Learn more about interacting with Pebble at at https://juju.is/docs/sdk/pebble.
-
-        Args:
-            event: event triggering the handler.
+        Raises:
+            RuntimeError: When the haproxy service is not running after install.
         """
-        # Get a reference the container attribute on the PebbleReadyEvent
-        container = event.workload
-        # Add initial Pebble config layer using the Pebble API
-        container.add_layer("httpbin", self._pebble_layer, combine=True)
-        # Make Pebble reevaluate its plan, ensuring any services are started if enabled.
-        container.replan()
-        # Learn more about statuses in the SDK docs:
-        # https://juju.is/docs/sdk/constructs#heading--statuses
+        self.haproxy_service.install()
```

Comment:
What about installing, enabling and if it is not running starting?
Something like https://github.com/canonical/github-runner-operator/blob/main/src/logrotate.py#L95

Reply:
Yep, I updated this to enable and start the service after install.

Reply:
So after discussing this with @gregory-schiano I decided that wrapping systemd error doesn't add much more benefits than simply letting the exception flow naturally to the top.

I also removed the calls to systemd.service_enable since we can be pretty confident that the service would be enabled after installing the apt package, and this behaviour is not likely to change with the stable release of haproxy.

I'm reopening this to continue the discussion here if needed.